### DISCUSSION TITLE
How to build/use an Example instance combined con Scorer?

### DISCUSSION THREAD

**Discussion by SergioMarreroMarrero** [explosion/spaCy#7797] [Help: Coding & Implementations] [answered]
Hi.

 I have a batch of documents that I want to classify (binary classification for example).  How to transform them in a batch of Annotated Examples that can be called from the Scorer?. 
 
 I think that just reading the documentation (API) of `Example` and `Scorer` is not enough to learn to implement them. Does anywhere an example or a little use-cases example about how to implement? I searched a lot on internet but I didn't find anything over there. I would appreciate a lot a little "tutorial" (some lines of code) about how to implement them.  

Thanks in advance.

**Comment by adrianeboyd** [comment, marked as answer]
For `doc.cats`, creating examples can look like this:

```python
import spacy
from spacy.training import Example
nlp = spacy.load("my_textcat_model")

texts = ["text1", "text2", "text3", ...]
categories = [{"CAT1": 1.0}, {"CAT2": 1.0}, {"CAT3": 1.0}, ...]

examples = []
for text, cats in texts, categories:
    example = Example.from_dict(nlp.make_doc(text), {"cats": cats})
    examples.append(example)

scores = nlp.evaluate(examples)
```

Or if you have annotated gold docs in `DocBin` as `file.spacy`, you can load them using the `Corpus` class:

```python
import spacy
from spacy.training import Corpus

nlp = spacy.load("my_textcat_model")

corpus = Corpus("file.spacy")
examples = list(corpus(nlp))
```

**Comment by SergioMarreroMarrero** [comment]
Hi.

Someone reading this question could feel interest on continue reading this. Hope it helps.

https://github.com/explosion/spaCy/discussions/7796

Greetings

**Comment by morankyle** [comment]
How would this code change for my use case? 

I'm trying to use nlp.evaluate for a custom NER model, so there is one blob of text with many different classes.

**Reply by polm** [reply]
Just to keep everything connected, note that I guess you opened https://github.com/explosion/spaCy/discussions/8024 to ask about this (so no reason to keep discussing it in this thread).
